sdk/monitor/Azure.Monitor.Ingestion/tests/ErrorTest.cs

```diff
@@ -107,5 +110,80 @@ public void TwoFailures()
                 Assert.AreEqual(413, exception.Status);
             }
         }
+
+        [Test]
+        public async Task OneFailureWithEventHandler()
+        {
+            LogsIngestionClient client = CreateClient();
+            // set compression to gzip so SDK does not gzip data (assumes already gzipped)
+            LogsIngestionClient.Compression = "gzip";
+            var entries = GenerateEntries(800, Recording.Now.DateTime);
+            entries.Add(new object[] {
+                    new {
+                        Time = Recording.Now.DateTime,
+                        Computer = "Computer" + new string('*', Mb),
+                        AdditionalContext = 1
+                    }
+                });
+
+            // Make the request
+            UploadLogsOptions options = new UploadLogsOptions();
+            var cts = new CancellationTokenSource();
+            bool isTriggered = false;
+            options.UploadFailedEventHandler += Options_UploadFailed;
+            //await client.UploadAsync(TestEnvironment.DCRImmutableId, TestEnvironment.StreamName, entries, options, cts.Token).ConfigureAwait(false);
+            await client.UploadAsync(TestEnvironment.DCRImmutableId, TestEnvironment.StreamName, entries, options).ConfigureAwait(false);
+            Assert.IsTrue(isTriggered);
+            Task Options_UploadFailed(UploadFailedEventArgs e)
+            {
+                isTriggered = true;
+                Assert.IsInstanceOf<RequestFailedException>(e.Exception);
+                Assert.AreEqual("ContentLengthLimitExceeded", ((RequestFailedException)(e.Exception)).ErrorCode);
+                Assert.IsNull(((RequestFailedException)(e.Exception)).InnerException);
+                Assert.AreEqual(413, ((RequestFailedException)(e.Exception)).Status);
+                //cts.Cancel();
+                return Task.CompletedTask;
+            }
+        }
+
+        [Test]
+        public async Task TwoFailuresWithEventHandler()
```

Comment:
It would be good to also have a test that covers the "error handler throwing" scenario.

Reply:
created issue to be addressed before GA: https://github.com/Azure/azure-sdk-for-net/issues/33699